docs/modules/ROOT/pages/explanations/jsonnet.adoc

```diff
@@ -0,0 +1,91 @@
+= Jsonnet Best Practices
+
+== Merging External Config Data
+////
+* Deep merge helper ✔
+* Merge from hiera ✔
+* Expose defaults in hiera ✔
+* Dict with key=name (too keep them mergeable) ✔
+* Provide "generic" defaults and merge with context specifics (i.e. cloud/region see openshift4-nodes) ✔
+* Allow Null (to enable empty) ✔
+* For non-mergeable data (i.e. strings), provide defaults + extras ✔
+////
+If a component manages an external configuration it's considered best practice to expose this configuration in the hierarchy and to define a sensible default where possible.
+This allows to override the configuration via the hierarchy.
+
+If multiple instances of a configuration are required, expose them within a dictionary where the keys correspond to the name of an instance.
+This allows for very flexible use of a component without changing any code.
+
+Allow to set a common configuration which is used as baseline for the multiple instances.
+This allows to reduce duplication by combining common options.
+
+Make sure to allow `Null` values for such dicts to support disabling the component by setting the config dict to `Null`.
+This usually requires an explicit check like the following example:
+[source,jsonnet]
+----
+local ingressControllers =
+  if params.ingressControllers != null then
+    std.objectFields(params.ingressControllers)
+  else
+    [];
+----
+
+The Commodore standard library provides a helper function `makeMergeable(o)` to make an object deep mergeable.
```

Comment:
For somebody not familiar with the subject, this does not make sense. Please provide context by explaining the behavior of keys ending with `+` and the fact that such keys can not be defined in yaml.